tsql/tsql.g4

```diff
@@ -695,6 +695,7 @@ expression
     | '~' expression                                           #unary_operator_expression
 
     | expression op=('*' | '/' | '%') expression               #binary_operator_expression
+    | expression op='||' expression                            #string_connection_expression
```

Comment:
Are you sure that T-SQL supports string connection operator? My SQL Management Studio notify about error `Incorrect syntax near '|'.` near such syntax: 
```SQL
select 'a' || 'b'
```